### PR TITLE
plumbum 1.9.0 ❄️

### PR DESCRIPTION
**Pull request by msentissi**
plumbum 1.9.0 ❄️

**Destination channel:** Snowflake

### Links

- [PKG-5958](https://anaconda.atlassian.net/browse/PKG-5958) 
- [Upstream repository](https://github.com/tomerfiliba/plumbum/tree/v1.9.0)

### Explanation of changes:

- Upgraded to v1.9.0 and added upstream tests

[PKG-5958]: https://anaconda.atlassian.net/browse/PKG-5958?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
   run:
-    - python >=3.6
-    - pywin32-on-windows
-    - paramiko
+    - python
+    - pywin32                     # [win]
+    - importlib_resources         # [py<39]
+  run_constrained:
+    - sphinx >=4.0.0
+    - sphinx-rtd-theme >=1.0.0
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
```
These are for building the upstream docs and are not relevant to the library itself.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
-{% set version = "1.8.1" %}
+{% set version = "1.9.0" %}
+{% set tests_to_skip = "" %}
+{% set tests_to_skip = tests_to_skip + "test_cwd or test_mixing_chdir or test_contains or 
+                                        test_path or test_iter_lines_buffer_size" %}
+{% set tests_to_skip = tests_to_skip + " or test_runfile_rich" %}       # [linux]
+{% set tests_to_skip = tests_to_skip + " or test_help_lang" %}          # [osx]
+{% set tests_to_skip = tests_to_skip + " or test_nolang_switches" %}    # [osx]
+{% set tests_to_skip = tests_to_skip + " or test_failure" %}            # [osx]
+{% set tests_to_skip = tests_to_skip + " or test_home" %}               # [win]
```
Comment (cbouss) [inline recipe/meta.yaml:10]: Not you have to change it, but you wouldn't have to skip test_home if you'd patch https://github.com/tomerfiliba/plumbum/blob/master/plumbum/_testtools.py#L16 adding `or sys.platform == "cygwin"`
https://github.com/tomerfiliba/plumbum/blob/v1.9.0/tests/test_env.py#L53

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,29 +1,42 @@
 {% set name = "plumbum" %}
-{% set version = "1.8.1" %}
+{% set version = "1.9.0" %}
+{% set tests_to_skip = "" %}
+{% set tests_to_skip = tests_to_skip + "test_cwd or test_mixing_chdir or test_contains or 
+                                        test_path or test_iter_lines_buffer_size" %}
+{% set tests_to_skip = tests_to_skip + " or test_runfile_rich" %}       # [linux]
```
Comment (cbouss) [inline recipe/meta.yaml:6]: Most failures seem to come from setting local on our builders. Ok to skip.

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (msentissi) [reply recipe/meta.yaml:10]: Nice catch thanks ! I'll keep the use of "cygwin" in mind for future windows issues.

**Pull request merged**